scripts/check_version

```diff
@@ -36,13 +56,36 @@ if [[ x${STRIPPED_PACKAGE_VERSION}x != x${IOS_MARKETING_VERSION_2}x ]]; then
 	EXITCODE=-1
 fi
 
-# This will check the format of IOS' CURRENT_PROJECT_VERSION
-if [[ ! "$IOS_PROJECT_VERSION_1" =~ ^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$ ]]; then
-    echo "Invalid format (expected X.X.X) for first CURRENT_PROJECT_VERSION (${IOS_PROJECT_VERSION_1}) on ios/HathorMobile.xcodeproj/project.pbxproj"
+# Check the format of iOS' CURRENT_PROJECT_VERSION
+# If it's a release version (without hyphen in package version), it should be 1.0.0
+# If it's a release candidate version (with hyphen in package version), it should be 0.rc.0, where rc is taken from package version
+if [ -z "$STRIPPED_PACKAGE_CANDIDATE" ]; then
+  if [[ x${IOS_PROJECT_VERSION_1}x != x1.0.0x ]]; then
+      echo "Invalid format (expected 1.0.0) for first CURRENT_PROJECT_VERSION (${IOS_PROJECT_VERSION_1}) on ios/HathorMobile.xcodeproj/project.pbxproj"
+      EXITCODE=-1
+  fi
+
+  if [[ x${IOS_PROJECT_VERSION_2}x != x1.0.0x ]]; then
+      echo "Invalid format (expected 1.0.0) for second CURRENT_PROJECT_VERSION (${IOS_PROJECT_VERSION_2}) on ios/HathorMobile.xcodeproj/project.pbxproj"
+      EXITCODE=-1
+  fi
+else
+  RCVERSION=${STRIPPED_PACKAGE_CANDIDATE//[^0-9]/}
+#  echo "RCVERSION: $RCVERSION"
+  if [[ x${IOS_PROJECT_VERSION_1}x != x0.${RCVERSION}.0x ]]; then
+      echo "Invalid format (expected 0.$RCVERSION.0) for first CURRENT_PROJECT_VERSION (${IOS_PROJECT_VERSION_1}) on ios/HathorMobile.xcodeproj/project.pbxproj"
+      EXITCODE=-1
+  fi
+
+  if [[ x${IOS_PROJECT_VERSION_2}x != x0.${RCVERSION}.0x ]]; then
+      echo "Invalid format (expected 0.$RCVERSION.0) for first CURRENT_PROJECT_VERSION (${IOS_PROJECT_VERSION_2}) on ios/HathorMobile.xcodeproj/project.pbxproj"
+      EXITCODE=-1
+  fi
 fi
 
-if [[ ! "$IOS_PROJECT_VERSION_2" =~ ^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$ ]]; then
-    echo "Invalid format (expected X.X.X) for second CURRENT_PROJECT_VERSION (${IOS_PROJECT_VERSION_2}) on ios/HathorMobile.xcodeproj/project.pbxproj"
+if [[ x${EXITCODE}x != x0x ]]; then
+  echo "Check version failed. Consider using 'make bump updateType=xx' to update the version"
```

Comment:
[suggestion] Why not present the failure after this message? I believe it would be better to keep messages in a concise and coherent place.

Reply:
The way this script is written, all error messages are written to `stdout`. This will be the last error message added in case of a failure. For example:

```
$ ./scripts/check_version
Version different in package.json (0.26.1) and android/app/build.gradle
Version different in package.json (0.26.1) and first MARKETING_VERSION (0.26.7) on ios/HathorMobile.xcodeproj/project.pbxproj
Invalid format (expected 1.0.0) for second CURRENT_PROJECT_VERSION (0.1.0) on ios/HathorMobile.xcodeproj/project.pbxproj
Check version failed. Consider using 'make bump updateType=xx' to update the version
```